include/misc/util.h

```diff
@@ -27,6 +27,13 @@
 #define POINTER_TO_INT(x)  ((s32_t) (x))
 #define INT_TO_POINTER(x)  ((void *) (x))
 
+#define BITS_PER_LONG	(__CHAR_BIT__) * (__SIZEOF_LONG__)
```

Comment:
Maybe this could be added just to help debugging a bit in case someone ever runs this with a different compiler:

```c
#if !(defined __CHAR_BIT__ || defined __SIZEOF_LONG__)
#	error Missing required predefined macros for BITS_PER_LONG calculation
#endif
```

Reply:
Yes! Good idea.